test/services/database/challenge_repository_test.dart

```diff
@@ -0,0 +1,252 @@
+import "package:cloud_firestore/cloud_firestore.dart";
+import "package:fake_cloud_firestore/fake_cloud_firestore.dart";
+import "package:flutter_test/flutter_test.dart";
+import "package:proxima/models/database/challenge/challenge_data.dart";
+import "package:proxima/models/database/challenge/challenge_firestore.dart";
+import "package:proxima/models/database/post/post_data.dart";
+import "package:proxima/models/database/post/post_firestore.dart";
+import "package:proxima/models/database/post/post_id_firestore.dart";
+import "package:proxima/models/database/user/user_firestore.dart";
+import "package:proxima/services/database/challenge_repository_service.dart";
+import "package:proxima/services/database/post_repository_service.dart";
+
+import "../../mocks/data/challenge_data.dart";
+import "../../mocks/data/firestore_user.dart";
+import "../../mocks/data/geopoint.dart";
+import "../../mocks/data/post_data.dart";
+
+/// Add [n] posts at position [pos] and return their data and the [PostFirestore] objects
+Future<List<PostFirestore>> addPostsFull(
```

Comment:
I personally find this function name ambiguous. I'd suggest `addPostsReturnFullPost` (courtesy of Copilot) for instance. This may make sense to rename `addPost` to `addPostsReturnPostData` too.